packages/babel-jest/src/index.ts

```diff
@@ -65,33 +66,54 @@ const createTransformer = (
     return babelConfig;
   }
 
+  const rootDirRegExpCache: {[rootDir: string]: RegExp} = Object.create(null);
+
+  function getRootDirRegExp(rootDir: Config.Path): RegExp {
+    if (!rootDirRegExpCache[rootDir]) {
+      // "/path/to/root/dir" or
+      // "path/to/root/dir/inner
+      rootDirRegExpCache[rootDir] = new RegExp(
+        `(['"])${escapeStrForRegex(rootDir)}(\\1|${escapeStrForRegex(
+          path.sep,
+        )})`,
+        'g',
+      );
+    }
+
+    return rootDirRegExpCache[rootDir];
+  }
+
   const transformer: BabelJestTransformer = {
     canInstrument: true,
     createTransformer,
     getCacheKey(
       fileData,
       filename,
-      configString,
+      _configString,
       {config, instrument, rootDir},
     ) {
       const babelOptions = loadBabelConfig(config.cwd, filename);
-      const configPath = [
-        babelOptions.config || '',
-        babelOptions.babelrc || '',
-      ];
+      const configPath = [babelOptions.config, babelOptions.babelrc].map(
+        configPath => configPath && path.relative(rootDir, configPath),
+      );
 
       return crypto
         .createHash('md5')
         .update(THIS_FILE)
         .update('\0', 'utf8')
-        .update(JSON.stringify(babelOptions.options))
+        .update(
+          // Not pretty but not coupled to a specific signature of the
+          // babel options to relativize paths
+          JSON.stringify(babelOptions.options).replace(
+            getRootDirRegExp(rootDir),
+            '$1<rootDir>$2',
+          ),
+        )
         .update('\0', 'utf8')
         .update(fileData)
         .update('\0', 'utf8')
         .update(path.relative(rootDir, filename))
         .update('\0', 'utf8')
-        .update(configString)
-        .update('\0', 'utf8')
```

Comment:
So now the cache won't be invalidated when we e.g. change some plugins in the babel config?

Reply:
it will, that's `babelOptions.options`. `configString` is essentially `JSON.stringify(projectConfig)`

Reply:
We were basically using `configString` as a catch-all but we were actually listing the things that actually impacted the transformation explicitly here (babel configuration and instrumentation).